internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,228 @@
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+type Rule = aggregationv1.KeyerConfiguration_Fragment_Rule
+
+type Mapper interface {
+	GetKeys(node core.Node, typeURL string) (string, error)
```

Comment:
Singular `GetKey`, since any request will only map to the first matched aggregated key.